R/paginate_listing.R

```diff
@@ -0,0 +1,92 @@
+#' Paginate listings
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'   Pagination of a listing. This can be vertical for long listings with many
+#'   rows or horizontal if there are many columns.
+#'
+#' @param lsting (`listing_df`)\cr the listing to paginate.
+#' @inheritParams formatters::pag_indices_inner
+#' @inheritParams formatters::vert_pag_indices
+#' @param lpp (`number`)\cr number of row lines (not counting titles and
+#'   footers) to have per page.
+#' @param colwidths (`numeric`)\cr print widths of columns, if manually
+#'   set/previously known.
+#'
+#' @returns A list of listings' objects that are meant to be on separated pages.
+#'
+#' @examples
+#' # Create a standard listing
+#' library(tibble)
+#' dat <- ex_adae
+#'
+#' lsting <- as_listing(dat[1:25, ],
+#'   key_cols = c("USUBJID", "AGE", "AESOC")
+#' ) %>%
+#'   add_listing_col("AETOXGR") %>%
+#'   add_listing_col("BMRKR1", format = "xx.x") %>%
+#'   add_listing_col("AESER / AREL", fun = function(df) paste(df$AESER, df$AREL, sep = " / "))
+#'
+#' # Vertical pagination
+#' paginate_listing(lsting, lpp = 10)
+#'
+#' # Horizontal pagination
+#' # paginate_listing(lsting, cpp = 10, lpp = 40)
+#'
+#' @export
+paginate_listing <- function(lsting, lpp = 15,
+                             cpp = NULL,
+                             min_siblings = 2,
+                             nosplitin = character(),
+                             colwidths = NULL,
+                             verbose = FALSE) {
+  # Input checks
+  checkmate::assert_count(lpp)
```

Comment:
`NULL` is an allowed value for `lpp`, even though its non-null by default. NULL means no vertical pagination.

Reply:
I did not know. Fixed along with the documentation